hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallerContext.java

```diff
@@ -36,6 +36,7 @@
 @InterfaceStability.Evolving
 public final class CallerContext {
   public static final Charset SIGNATURE_ENCODING = StandardCharsets.UTF_8;
+  public static final String ITEM_SEPARATOR = "\\$";
```

Comment:
This was the issue that Daryn was bringing up.
I'm not sure if overriding an existing concept is safe.
The idea would be to add a new one with the proper key/value semantic.

Reply:
Have made it configurable